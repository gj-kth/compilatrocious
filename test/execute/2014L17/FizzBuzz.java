/*
 * FizzBuzz without strings!
 */

class Main1 {
    public static void main(String[] a) {
        FizzBuzz fizzbuzzer;
        int i;
        fizzbuzzer = new FizzBuzz();
        i = 1;

        while (i < 100) {
            if (fizzbuzzer.test(i, 15)) {
                System.out.println(51228022);
            }
            else if (fizzbuzzer.test(i, 5)) {
                System.out.println(8022);
            }
            else if (fizzbuzzer.test(i, 3)) {
                System.out.println(5122);
            }
            else {
                System.out.println(i);
            }
            i = i + 1;
        }
    }
}

class FizzBuzz {

    public boolean test(int l, int r) {
        int v;
        v = this.modulo(l, r);
        return v < 1 && (0 - 1) < v;
    }
    
    public int modulo(int l, int r) {
        while (r < l + 1) {
            l = l - r;
        }
        return l;
    }
}
