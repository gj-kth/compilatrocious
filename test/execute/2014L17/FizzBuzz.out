1
2
5122
4
8022
5122
7
8
5122
8022
11
5122
13
14
51228022
16
17
5122
19
8022
5122
22
23
5122
8022
26
5122
28
29
51228022
31
32
5122
34
8022
5122
37
38
5122
8022
41
5122
43
44
51228022
46
47
5122
49
8022
5122
52
53
5122
8022
56
5122
58
59
51228022
61
62
5122
64
8022
5122
67
68
5122
8022
71
5122
73
74
51228022
76
77
5122
79
8022
5122
82
83
5122
8022
86
5122
88
89
51228022
91
92
5122
94
8022
5122
97
98
5122
